pkg/xen-tools/patches-4.14.0/13-qemu-Revamp-mmap-allocation-routine.patch

```diff
@@ -0,0 +1,124 @@
+From 3ad5ce46a00385fc59adb5ba9f7c4b358e346bbf Mon Sep 17 00:00:00 2001
+From: Sergey Temerkhanov <s.temerkhanov@gmail.com>
+Date: Sun, 20 Sep 2020 02:44:45 +0300
+Subject: [PATCH 2/2] qemu: Revamp mmap allocation routine
+
+Do not double-map the memory addresses
+
+Signed-off-by: Sergey Temerkhanov <s.temerkhanov@gmail.com>
+---
+ util/mmap-alloc.c | 42 +++++++++++++++---------------------------
+ 1 file changed, 15 insertions(+), 27 deletions(-)
+
+diff --git a/util/mmap-alloc.c b/util/mmap-alloc.c
+index 27dcccd8ec..7ff2883e42 100644
+--- a/tools/qemu-xen/util/mmap-alloc.c
++++ b/tools/qemu-xen/util/mmap-alloc.c
+@@ -89,19 +89,18 @@ void *qemu_ram_mmap(int fd,
+                     bool is_pmem)
+ {
+     int flags;
++    int mapfd;
+     int map_sync_flags = 0;
+-    int guardfd;
+     size_t offset;
+     size_t pagesize;
+     size_t total;
+-    void *guardptr;
+     void *ptr;
++    int ret;
+ 
+     /*
+      * Note: this always allocates at least one extra page of virtual address
+      * space, even if size is already aligned.
+      */
+-    total = size + align;
```

Comment:
Why are we dropping the requirement of mapping an extra page? This seems pretty negligible in the grand scheme of things, no?

Reply:
We're  doing it a different way - we allocate the whole region and then set these guard pages via mprotect()